### ISSUE TITLE
addon does not seem to work on Firefox Mobile

### ISSUE THREAD

**Issue by ghost** [proginosko/LeechBlockNG#301]
Can't seem to get blocking to work on firefox mobile.
The timer shows up in the top left corner but nothing happens.
Everything is updated to the latest version.
I recreated my settings on desktop and it works fine.

Is this just a mobile thing?

**Comment by proginosko** [comment]
As far as I know, it still works on mobile. Can you email me screenshots of your settings? leechblockng@proginosko.com

**Comment by leo1115** [comment]
Hi were you able to get it to work. i have similar issue.

**Comment by proginosko** [comment]
Again, I need more details in order for me to investigate this.

**Comment by sha-265** [comment]
For me it's not working properly on stable version of Firefox for Android. In Firefox Beta it's working as expected. I use the same LB config on both.

**Comment by proginosko** [comment]
> For me it's not working properly on stable version of Firefox for Android.

Is it not working at all, or only partly (and which parts)?

**Comment by sha-265** [comment]
> > For me it's not working properly on stable version of Firefox for Android.
> 
> Is it not working at all, or only partly (and which parts)?

Partly. For example I have a set who set grayscale filter on some domain, and it's not always working, sometimes it does sometimes, no.

**Comment by proginosko** [comment]
Okay, thanks. So it works fine on Firefox Beta?

**Comment by sha-265** [comment]
> Okay, thanks. So it works fine on Firefox Beta?

Yes.

**Comment by krin-san** [comment]
Blocking with mostly default settings doesn't work on Firefox 122.0 on Fire HD 10 2019 installed though Google Play Store side-loaded with Fire Toolbox. Yeah, quite a hacky environment, but I see no reason why website blocking won't work. Especially given the fact that **it actually works** if `Close tab instead of blocking page` is enabled.

# Differences with default config

Number of Block Sets: 1

What to block:
- Custom name: "Main"
- Domains:
```
*
+kikaninchen.de
```

When to block:
- All Day
- ☑️ Selected all days of the week

How to Block:
- ☑️ Immediately block pages ...
- ☑️ Add blocked page to browser history
---

**Comment by proginosko** [comment]
What happens if you deselect "Add blocked page to browser history"? I just realized that Firefox for Android doesn't support the [history API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/history), so I almost certainly the blocking function is generating an error when LB tries to add the blocked URL to the browser history and bombing out before the blocking page is loaded. The "Add blocked page to browser history" is actually ignored (probably wrongly, but there it is!) when the "Close tab" option is used.

**Comment by sha-265** [comment]
This option is disable in my case, and still have problems.

**Comment by proginosko** [comment]
> This option is disabled in my case, and still have problems.

I think the problems you are experiencing are different than those of @krin-san. Yours have to do with a bug in the implementation of Manifest V3 for Firefox for Android (see [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1872600)).

**Comment by krin-san** [comment]
@proginosko you're totally right! When `Add blocked page to browser history` and `Close tab instead of blocking page` options are disabled, LeechBlock shows the "blocked" page just as expected!

**Comment by proginosko** [comment]
@krin-san Thanks for confirming. I guess I'll need to hide/ignore the add-to-history option for Android users in future versions.
